src/escrows/DolaFraxBPEscrow.sol

```diff
@@ -0,0 +1,271 @@
+        if (yearn.balanceOf(address(this)) > 0)
+            revert CannotDepositToConvex(yearn.balanceOf(address(this)));
+        uint256 tokenBal = token.balanceOf(address(this));
+        stakedBalance += tokenBal;
+        booster.deposit(pid, tokenBal, true);
+    }
+
+    /**
+     * @notice Deposit token balance in the escrow into Yearn
+     * @dev Cannot deposit if there are Convex tokens in the escrow (only 1 strategy at a time)
+     */
+    function depositToYearn() external onlyBeneficiary {
+        if (stakedBalance > 0) revert CannotDepositToYearn(stakedBalance);
+        yearn.deposit(token.balanceOf(address(this)), address(this));
+    }
+
+    /**
+     * @notice Withdraw all tokens from Yearn
+     * @return lpAmount The amount of tokens withdrawn from Yearn
+     */
+    function withdrawFromYearn()
+        external
+        onlyBeneficiary
+        returns (uint256 lpAmount)
+    {
+        return _fullWithdrawFromYearn();
+    }
+
+    /**
+     * @notice Withdraw all tokens from Convex
+     * @return lpAmount The amount of tokens withdrawn from Convex
+     */
+    function withdrawFromConvex()
+        external
+        onlyBeneficiary
+        returns (uint256 lpAmount)
+    {
+        return _fullWithdrawFromConvex();
+    }
+
+    /**
+     * @notice Move all tokens deposited into Convex to Yearn
+     * @dev Will move all tokens from Convex to Yearn, including extra lp tokens that might be in the escrow if useAll is true
+     * @param useAll If true, deposit the full balance in the escrow
+     * @return lpAmount The amount of tokens deposited into Yearn
+     */
+    function moveFromConvexToYearn(
+        bool useAll
+    ) external onlyBeneficiary returns (uint256 lpAmount) {
+        lpAmount = _fullWithdrawFromConvex();
+
+        if (useAll) lpAmount = token.balanceOf(address(this));
+
+        yearn.deposit(lpAmount, address(this));
+    }
+
+    /**
+     * @notice Move all tokens deposited into Yearn to Convex
+     * @dev Will move all tokens from Yearn to Convex, including extra lp tokens that might be in the escrow if useAll is true
+     * @param useAll If true, deposit the full balance in the escrow
+     * @return lpAmount The amount of tokens deposited into Convex
+     */
+    function moveFromYearnToConvex(
+        bool useAll
+    ) external onlyBeneficiary returns (uint256 lpAmount) {
+        lpAmount = _fullWithdrawFromYearn();
+
+        if (useAll) lpAmount = token.balanceOf(address(this));
+
+        stakedBalance += lpAmount;
+        booster.deposit(pid, lpAmount, true);
+    }
+
+    function _fullWithdrawFromYearn() internal returns (uint256 lpAmount) {
+        lpAmount = yearn.withdraw(
+            yearn.balanceOf(address(this)),
+            address(this)
+        );
+    }
+
+    function _fullWithdrawFromConvex() internal returns (uint256 lpAmount) {
+        lpAmount = stakedBalance;
+        stakedBalance = 0;
+        rewardPool.withdrawAndUnwrap(lpAmount, false);
+    }
```

Comment:
Move logic to withdraw functions as withdraw functions are currently just empty wrappers for these.